--- firstboot-1.3.33/src/firstboot.py.race	2005-05-30 15:56:05.000000000 -0400
+++ firstboot-1.3.33/src/firstboot.py	2005-05-30 15:56:48.000000000 -0400
@@ -197,7 +197,12 @@
                 raise RuntimeError, "X server failed to start"
             try:
                 os.stat("/tmp/.X11-unix/X1")
-                break
+                try:
+                    import gtk
+                    break
+                except Exception, e:
+                    print "error connecting to X server:", str(e)
+                    print "will retry in 1 second"
             except OSError:
                 pass
             time.sleep(1)
